### PR TITLE
Captured Tensorboard log in autologging for tf.Estimator APIs

### PR DESCRIPTION
**Pull request by jinzhang21**
## What changes are proposed in this pull request?
Added support for Tensorboard log in autologging with tf.Estimator.train() API.

## How is this patch tested?
1. Added 2 unit tests.
2. Tested and verified locally via mlflow ui.

## Release Notes
Training via Tensorflow Estimator APIs now saves Tensorboard logs in MLFlow tracking when autologging is enabled.

### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [x] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [x] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: Local serving, model deployment tools, spark UDFs
- [ ] `area/server-infra`: MLflow server, JavaScript dev server
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, JavaScript, plotting
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/tensorflow.py [hunk 1]
```diff
@@ -912,7 +912,15 @@ def train(original, self, *args, **kwargs):
             try_mlflow_log(mlflow.log_param, "max_steps", kwargs["max_steps"])
 
         result = original(self, *args, **kwargs)
-
+        # Logs Tensorboard to artifacts.
```
Comment (dbczumar) [inline mlflow/tensorflow.py:915]: Minor suggestion for inline comment:

```suggestion
        # Logs Tensorboard event files as artifacts
```

File: mlflow/tensorflow.py [hunk 2]
```diff
@@ -912,7 +912,15 @@ def train(original, self, *args, **kwargs):
...
+        for file in os.listdir(self.model_dir):
```
Comment (dbczumar) [inline mlflow/tensorflow.py:916]: can we handle the case where `self.model_dir` does not exist?

**Review by dbczumar — COMMENTED** [review]
@jinzhang21 looks great in general! Left a couple small comments. Looks like all tests pass on the PR except for the DCO check. Here are some instructions for getting that one to pass: https://github.com/mlflow/mlflow/pull/4090/checks?check_run_id=1881244252

File: mlflow/tensorflow.py (hunk 2 above)
Comment (jinzhang21) [reply mlflow/tensorflow.py]: Done.

**Review by dbczumar — APPROVED** [review]
LGTM! Thanks a bunch @jinzhang21 !

**Pull request merged**